--- client.py
+++ client.py
@@ -42,6 +42,7 @@
 from copy import deepcopy
 from suds.plugin import PluginContainer
 from logging import getLogger
+from xml.sax import SAXParseException
 
 log = getLogger(__name__)
 
@@ -699,9 +710,13 @@
         log.debug('http failed:\n%s', reply)
         if status == 500:
             if len(reply) > 0:
-                r, p = binding.get_fault(reply)
-                self.last_received(r)
-                return (status, p)
+                try:
+                    r, p = binding.get_fault(reply)
+                except SAXParseException:
+                    return (status, None)
+                else:
+                    self.last_received(r)
+                    return (status, p)
             else:
                 return (status, None)
         if self.options.faults:
